src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCallerEngine.java

```diff
@@ -294,16 +284,15 @@ private void initialize(boolean createBamOutIndex, final boolean createBamOutMD5
 
         initializeActiveRegionEvaluationGenotyperEngine();
 
-        defaultGenotypingEngine = new HaplotypeCallerGenotypingEngine(hcArgs, samplesList, ! hcArgs.doNotRunPhysicalPhasing, hcArgs.applyBQD);
-        defaultGenotypingEngine.setAnnotationEngine(annotationEngine);
-
-        // Create other custom genotyping engines if user provided ploidyRegions
-        for (final int ploidy : this.allCustomPloidies) {
-            HaplotypeCallerArgumentCollection newPloidyHcArgs = hcArgs.copyWithNewPloidy(ploidy);
-            HaplotypeCallerGenotypingEngine newGenotypingEngine = new HaplotypeCallerGenotypingEngine(newPloidyHcArgs, samplesList, ! hcArgs.doNotRunPhysicalPhasing, hcArgs.applyBQD);
-            newGenotypingEngine.setAnnotationEngine(annotationEngine);
-            this.ploidyToGenotyperMap.put(ploidy, newGenotypingEngine);
-        }
+        genotypingEngines = new MultiPloidyGenotyper<>((ploidy) -> {
+                    final HaplotypeCallerArgumentCollection newPloidyHcArgs = hcArgs.copyWithNewPloidy(ploidy);
```

Comment:
So this refactor still assumes a solution to copy the `hcArgs`? I might end up trying to refactor in a way to avoid that like we discussed, but we can keep this here for now.

Reply:
After chatting with David, I'll end up keeping this method implemented, but using the Apache clone ser/de method.